src/DotNetWorker.Grpc/AppContextUtils.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.Functions.Worker.Grpc
+{
+    internal class AppContextUtils
+    {
+        /// <summary>
+        /// Returns a boolean value indicating whether the worker is running with native host(placeholder support).
+        /// </summary>
+        internal static bool IsRunningWithNativeHost() => AppContext.GetData("AZURE_FUNCTIONS_NATIVE_HOST") is not null;
```

Comment:
Might be out of scope for this PR but between capabilities that already exist (and more props to come with this work + http proxying) we might want to make a constants file similar to [the host one](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Workers/Rpc/RpcWorkerConstants.cs).

Reply:
+1 to using a const

Reply:
Removed this helper in the new iteration based on the other feedback about conditionally setting those props.